model/offline/JsonOfflineTestImporter.php

```diff
@@ -203,32 +203,41 @@ protected function importActions()
         /** @var DeliveryExecutionInterface $deliveryExecution */
         $deliveryExecution = $this->getProxyService()->getDeliveryExecution($this->getOfflineParser()->getSessionId());
 
-        if ($deliveryExecution->getState()->getUri() != DeliveryExecutionInterface::STATE_ACTIVE) {
+        if ($deliveryExecution->getState()->getUri() !== DeliveryExecutionInterface::STATE_ACTIVE) {
             throw new \common_Exception(__('Delivery execution %s is not active state.', $deliveryExecution->getIdentifier()));
         }
 
         /** @var QtiRunnerServiceContext $serviceContext */
         $serviceContext = $this->getServiceContext($deliveryExecution);
 
-        /** @var AssessmentTestSession $testSession */
-        $testSession = $serviceContext->getTestSession();
+        $testState = DeliveryExecutionInterface::STATE_TERMINATED;
 
-        $testSession->setVariable(new OutcomeVariable(
-            OfflineTestParserInterface::IS_OFFLINE_VARIABLE,
-            Cardinality::SINGLE,
-            BaseType::FLOAT,
-            new QtiFloat(1.0))
-        );
-        $serviceContext->setTestSession($testSession);
+        if (!$this->getOfflineParser()->isInterrupted()) {
```

Comment:
i'd extracted `$this->getOfflineParser()` to variable cause it is used below one more time

Reply:
It always returns property, so I don't see any reason to avoid the double call of the method. Creating an additional variable looks worse in this case.

Reply:
However, I changed it to get more readable code.